monai/networks/nets/torchvision_fc.py

```diff
@@ -9,66 +9,56 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Tuple, Union
-
-import torch
+from typing import Any, Dict, Optional, Tuple
 
+from monai.networks.nets import NetAdapter
 from monai.utils import optional_import
 
 models, _ = optional_import("torchvision.models")
 
 
-__all__ = ["TorchVisionFullyConvModel"]
+__all__ = ["TorchVisionFCModel"]
 
 
-class TorchVisionFullyConvModel(torch.nn.Module):
+class TorchVisionFCModel(NetAdapter):
     """
-    Customize TorchVision models to replace fully connected layer by convolutional layer.
+    Customize the fully connected layer of TorchVision model or replace it by convolutional layer.
 
     Args:
-        model_name: name of any torchvision with adaptive avg pooling and fully connected layer at the end.
+        model_name: name of any torchvision model with fully connected layer at the end.
             ``resnet18`` (default), ``resnet34m``, ``resnet50``, ``resnet101``, ``resnet152``,
             ``resnext50_32x4d``, ``resnext101_32x8d``, ``wide_resnet50_2``, ``wide_resnet101_2``.
+            model details: https://pytorch.org/vision/stable/models.html.
         n_classes: number of classes for the last classification layer. Default to 1.
-        pool_size: the kernel size for `AvgPool2d` to replace `AdaptiveAvgPool2d`. Default to (7, 7).
-        pool_stride: the stride for `AvgPool2d` to replace `AdaptiveAvgPool2d`. Default to 1.
+        dim: number of spatial dimensions, default to 2.
+        in_channels: number of the input channels of last layer. if None, get it from `in_features` of last layer.
+        use_conv: whether use convolutional layer to replace the last layer, default to False.
+        pool: parameters for the pooling layer, it should be a tuple, the first item is name of the pooling layer,
+            the second item is dictionary of the initialization args. if None, will not replace the `layers[-2]`.
+            default to `("avg", {"kernel_size": 7, "stride": 1})`.
+        bias: the bias value when replacing the last layer. if False, the layer will not learn an additive bias,
+            default to True.
         pretrained: whether to use the imagenet pretrained weights. Default to False.
     """
 
     def __init__(
         self,
         model_name: str = "resnet18",
         n_classes: int = 1,
-        pool_size: Union[int, Tuple[int, int]] = (7, 7),
-        pool_stride: Union[int, Tuple[int, int]] = 1,
+        dim: int = 2,
+        in_channels: Optional[int] = None,
+        use_conv: bool = False,
+        pool: Optional[Tuple[str, Dict[str, Any]]] = ("avg", {"kernel_size": 7, "stride": 1}),
+        bias: bool = True,
```

Comment:
I think the layer type checks could be enabled here

Reply:
Do you mean to check that the last layer or both the last FC layer and pooling layer as before?

Thanks.

Reply:
BTW, is there any Torchvision classification model doesn't use FC as the final layer and we also want to support it?
If so, we should not check the layer type.

Thanks.

Reply:
yes, this model is from `getattr(models, model_name)`, so we can always check them

Reply:
Added logic to check the final FC layer in the latest commit.
Thanks.